src/main/kotlin/app/revanced/patches/music/layout/premium/patch/HideGetPremiumPatch.kt

```diff
@@ -0,0 +1,55 @@
+package app.revanced.patches.music.layout.premium.patch
+
+import app.revanced.patcher.annotation.Description
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.impl.BytecodeData
+import app.revanced.patcher.extensions.addInstructions
+import app.revanced.patcher.extensions.replaceInstruction
+import app.revanced.patcher.fingerprint.method.utils.MethodFingerprintUtils.resolve
+import app.revanced.patcher.patch.annotations.Patch
+import app.revanced.patcher.patch.impl.BytecodePatch
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patches.music.layout.premium.annotations.HideGetPremiumCompatibility
+import app.revanced.patches.music.layout.premium.fingerprints.HideGetPremiumFingerprint
+import app.revanced.patches.music.layout.premium.fingerprints.HideGetPremiumParentFingerprint
+import org.jf.dexlib2.builder.instruction.BuilderInstruction35c
+
+@Patch
+@Name("hide-get-premium")
+@Description("Removes all \"Get Premium\" evidences from the avatar menu.")
+@HideGetPremiumCompatibility
+@Version("0.0.1")
+class HideGetPremiumPatch : BytecodePatch(
+    listOf(
+        HideGetPremiumParentFingerprint
+    )
+) {
+    override fun execute(data: BytecodeData): PatchResult {
+        val parentResult = HideGetPremiumParentFingerprint.result!!
+        HideGetPremiumFingerprint.resolve(data, parentResult.classDef)
+
+        val method = HideGetPremiumFingerprint.result!!.mutableMethod
+
+        val firstIndex = method.implementation!!.instructions.indexOfFirst {
+            ((it as? BuilderInstruction35c)?.reference.toString() == "Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z")
+        }
+        method.replaceInstruction(
+            firstIndex - 6, """
```

Comment:
Is -6 not a bit too far? In the fingerprint `HideGetPremiumFingerprint` add a 1-x opcodes which match to the instructions you need to change.

Reply:
I'm not using opcodes on the fingerprint, I'm using a string... how can I achieve what you're asking?

Reply:
Check how other fingerprints are made with both an opcode pattern and strings

Reply:
I have to get the index of two places in this (quite huge) method so I'll split into two fingerprint classes to get each index. Is this the right thing to do?

Reply:
Theres multiple ways to do what you want to do

Reply:
You'd need to simply create a fingerprint to find the method, then a second one with just the opcode pattern and resolve it to the method. The second fingerprint will then contain the information for the scan result.

Reply:
Ok. But then I should have a third fingerprint with another opcode pattern to get the second place in the method which I want to retrieve. Does this make sense?

Reply:
No, that would be done by the first fingerprint already.

Reply:
Ok, so the first fingerprint should find the method and the first index which I want to change. The second fingerprint should find the second index alone, using the first fingerprint to resolve/find the method. Is this ok now?

Reply:
It's working with https://github.com/revanced/revanced-patches/pull/195/commits/f0f5450ea8080069b68fb61041018fd0dd89abfe! Thanks for the help!
Always great to learn from you both :)